File: legishub-service/wallet-service/src/main/java/br/com/cin/aps/apjcardservice/entity/Cartao.java
package br.com.cin.aps.apjcardservice.entity;

import javax.persistence.*;

@Entity
public class Cartao {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    Long id;

    @Column(name = "login_usuario")
    String loginUsuario;

    @Column
    @Enumerated
    BandeiraCartao bandeira;

    @Column
    String numero;

    @Column
    int codigo;

    @Column
    int validade;

    public Long getId() {return id;}

    public void setId(Long id) {this.id = id;}

    public String getLoginUsuario() {return loginUsuario;}

    public void setLoginUsuario(String loginUsuario) {this.loginUsuario = loginUsuario;}

    public BandeiraCartao getBandeira() {
        return bandeira;
    }

    public void setBandeira(BandeiraCartao cartao) {
        this.bandeira = cartao;
    }

    public String getNumero() {
        return numero;
    }

    public void setNumero(String numero) {
        this.numero = numero;
    }

    public int getCodigo() {
        return codigo;
    }

    public void setCodigo(int codigo) {
        this.codigo = codigo;
    }

    public int getValidade() {
        return validade;
    }

    public void setValidade(int validade) {
        this.validade = validade;
    }
}

File: legishub-service/apj-service/src/main/kotlin/br/com/cin/aps/apjservice/entity/Cartao.kt
package br.com.cin.aps.apjservice.entity

import java.math.BigInteger
import javax.persistence.Column
import javax.persistence.Entity
import javax.persistence.Id

@Entity
data class Cartao(@Column val bandeira : BandeiraCartao,
                  @Column val numero : BigInteger,
                  @Column val codigo: Int,
                  @Column val validade: Int) {
    @Id val id: Long? = null

}

class CreditCardBuilder{
    private var bandeira : BandeiraCartao? = null
    private var number : BigInteger? = null
    private var code: Int? = null
    private var validade: Int? = null

     fun flag(creditCardFlag: BandeiraCartao) : CreditCardBuilder{ bandeira = creditCardFlag ; return this}

     fun number(number: BigInteger) : CreditCardBuilder{ this.number=number ; return this}

     fun code(code: Int) : CreditCardBuilder{ this.code=code ; return this}
    fun validade(validade: Int) : CreditCardBuilder { this.validade=validade ; return this}

     fun build() : Cartao = Cartao(bandeira!!,number!!,code!!, validade!!)


}




enum class BandeiraCartao{
    VISA,MASTERCARD
}
